pegasus/sites.v3/hourofcode.com/public/learn/index.haml

```diff
@@ -21,22 +21,27 @@ social:
 %script{src: webpack_asset_path("js/#{js_locale}/tutorialExplorer_locale.js")}
 %script{src: webpack_asset_path("js/tutorialExplorer.js")}
 
+- tutorial_sheet_id = request.params["preview"] ? :tutorials_preview : :tutorials
+- no_cache = !!request.params["preview"]
+- tutorials_contents = Tutorials.new(tutorial_sheet_id, no_cache).contents(request.host_with_port)
+
 -# Load tutorial images from code.org so we don't have to duplicate them on hourofcode.com.
-- tutorials_contents = Tutorials.new(:tutorials).contents(request.host_with_port)
 - tutorials_contents.each do |tutorial|
   - tutorial[:image] = CDO.code_org_url(tutorial[:image])
 
 - tutorials_contents.each do |tutorial|
   - tutorial[:contact] = nil
   - prefix = "tutorial_#{tutorial[:short_code]}_"
-  - tutorial[:name]                                 = hoc_s(prefix + "name")
-  - tutorial[:shortdescription]                     = hoc_s(prefix + "shortdescription")
-  - tutorial[:longdescription]                      = hoc_s(prefix + "longdescription")
-  - tutorial[:language]                             = hoc_language(tutorial[:languages_supported])
-  - tutorial[:string_detail_grades]                 = hoc_s(prefix + "string_detail_grades")
-  - tutorial[:string_platforms]                     = hoc_s(prefix + "string_platforms")
-  - tutorial[:string_detail_platforms]              = hoc_s(prefix + "string_detail_platforms")
-  - tutorial[:string_detail_programming_languages]  = hoc_s(prefix + "string_detail_programming_languages")
+  -# Unless showing a preview, look up localized versions of strings.
+  - unless request.params["preview"]
```

Comment:
nit: can you pull this out into a variable, ie `preview = request.params["preview"]`. At least for me, that would be more readable

Reply:
Good call, will do.